File: Examples/Node.js/Images/ExtractParticularImageWithSpecifiedSize.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var PdfApi = require('asposepdfcloud');
var StorageApi = require('asposestoragecloud');
var configProps = require('../config/config.json');
var data_path = '../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose.Pdf API SDK
var pdfApi = new PdfApi(config);

// Set input file name
var fileName = "SampleImage";
var name = "SampleImage.pdf";
var pageNumber = 1;
var imageNumber = 1;
var format = "jpeg";
var width = 200;
var height = 200;

try {
// Upload source file to aspose cloud storage
storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {

	assert.equal(responseMessage.status, 'OK');

	// Invoke Aspose.Pdf Cloud SDK API to extract a particular image from a PDF page with specified size
	pdfApi.GetImageWithFormat(name, pageNumber, imageNumber, format, width, height, null, null, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			// Download image from response
			var outfilename = fileName + "." + format;
			var writeStream = fs.createWriteStream(data_path + outfilename);
			writeStream.write(responseMessage.body);
			});
	});

}
catch (e) 
{
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Perl/Images/ExtractParticularImageWithSpecifiedSize.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;

use AsposePdfCloud::PdfApi;
use AsposePdfCloud::ApiClient;
use AsposePdfCloud::Configuration;

my $configFile = '../config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposePdfCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposePdfCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposePdfCloud::Configuration::debug = 1;

$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Pdf API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $pdfApi = AsposePdfCloud::PdfApi->new();

# Set input file name
my $fileName =  'SampleImage';
my $name =  'SampleImage.pdf';
my $pageNumber =  1;
my $imageNumber =  1;
my $format ='jpeg';
my $width = 200;
my $height = 200;

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);

# Invoke Aspose.Pdf Cloud SDK API to extract a particular image from a PDF page with specified size                       
$response = $pdfApi->GetImageWithFormat(name=>$name, pageNumber=>$pageNumber, imageNumber=>$imageNumber, format=>$format, width=>$width, height=>$height);

if($response->{'Status'} eq 'OK'){
	# Download image from response
    my $output_file = $out_path. $fileName . '.' . $format;
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1